### PR TITLE
[WIP] refactor: use injectLocale in Upload

### PR DESCRIPTION
**Pull request by benjycui**
ping @bang88

### CODE REVIEW

File: components/upload/index.tsx [hunk 1]
```diff
@@ -2,4 +2,9 @@ import Upload from './Upload';
 import Dragger from './Dragger';
 
 Upload.Dragger = Dragger;
+
+function Test() {
+  return <Upload.Dragger />;
```
Comment (benjycui) [inline components/upload/index.tsx]: This doesn't work as I expected, `Upload.Dragger` is any, so that we cannot get any tips in VSCode.

Comment (bang88) [inline components/upload/index.tsx]: Yep, you are right , but we don't want hard code every static member.

File: components/locale-provider/injectLocale.tsx [hunk 2]
```diff
@@ -8,8 +8,12 @@ export interface ComponentContext {
   antLocale?: { [key: string]: any };
 }
 
+export interface ComponentClass<P> {
+  [key: string]: any;
```
Comment (bang88) [inline components/locale-provider/injectLocale.tsx]: If we don't extends from `React.ComponentClass<P>` will lose the default static members.

**Comment by bang88** [comment]
这个重构要不就先暂缓一下，等有好一点的解决方案再做，现在实现起来有点不舒服。

**Comment by benjycui** [comment]
目前的确没有好的办法，push 上来只是为了让你可以直观的看到问题而已。

**Pull request closed without merging**